src/main/java/bio/terra/service/policy/TerraPolicyService.java

```diff
@@ -0,0 +1,114 @@
+import bio.terra.model.RepositoryStatusModelSystems;
+import bio.terra.policy.api.PublicApi;
+import bio.terra.policy.api.TpsApi;
+import bio.terra.policy.client.ApiException;
+import bio.terra.policy.model.TpsComponent;
+import bio.terra.policy.model.TpsObjectType;
+import bio.terra.policy.model.TpsPaoCreateRequest;
+import bio.terra.policy.model.TpsPolicyInputs;
+import bio.terra.service.policy.exception.PolicyConflictException;
+import bio.terra.service.policy.exception.PolicyServiceApiException;
+import bio.terra.service.policy.exception.PolicyServiceAuthorizationException;
+import bio.terra.service.policy.exception.PolicyServiceDuplicateException;
+import bio.terra.service.policy.exception.PolicyServiceNotFoundException;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TerraPolicyService {
+  private static final Logger logger = LoggerFactory.getLogger(TerraPolicyService.class);
+  private final PolicyServiceConfiguration policyServiceConfiguration;
+  private final PolicyApiService policyApiService;
+
+  public TerraPolicyService(
+      PolicyServiceConfiguration policyServiceConfiguration, PolicyApiService policyApiService) {
+    this.policyServiceConfiguration = policyServiceConfiguration;
+    this.policyApiService = policyApiService;
+  }
+
+  // -- Policy Attribute Object Interface --
+  public void createSnapshotPao(UUID snapshotId, @Nullable TpsPolicyInputs policyInputs) {
+    createPao(snapshotId, TpsObjectType.SNAPSHOT, policyInputs);
+  }
+
+  public void createPao(
+      UUID resourceId, TpsObjectType resourceType, @Nullable TpsPolicyInputs policyInputs) {
+    policyServiceConfiguration.tpsEnabledCheck();
+    TpsPolicyInputs inputs = (policyInputs == null) ? new TpsPolicyInputs() : policyInputs;
+
+    TpsApi tpsApi = policyApiService.getPolicyApi();
+    try {
+      tpsApi.createPao(
+          new TpsPaoCreateRequest()
+              .objectId(resourceId)
+              .component(TpsComponent.TDR)
+              .objectType(resourceType)
+              .attributes(inputs));
+    } catch (ApiException e) {
+      throw convertApiException(e);
+    }
+  }
+
+  public void deletePao(UUID resourceId) {
+    policyServiceConfiguration.tpsEnabledCheck();
+    TpsApi tpsApi = policyApiService.getPolicyApi();
+    try {
+      try {
+        tpsApi.deletePao(resourceId);
+      } catch (ApiException e) {
+        throw convertApiException(e);
+      }
+    } catch (PolicyServiceNotFoundException e) {
+      // Not found should not cause a failure on delete.
+    }
+  }
+
+  @VisibleForTesting
+  static RuntimeException convertApiException(ApiException ex) {
+    if (ex.getCode() == HttpStatus.UNAUTHORIZED.value()) {
```

Comment:
If you use `org.apache.http.HttpStatus` instead you can convert this to an expression switch.
```
    return switch (e.getCode()) {
      case HttpStatus.SC_UNAUTHORIZED -> new PolicyServiceAuthorizationException(
          "Not authorized to access Terra Policy Service", e.getCause());
      case HttpStatus.SC_NOT_FOUND ->
          new PolicyServiceNotFoundException("Policy service returns not found exception", e);
      case HttpStatus.SC_BAD_REQUEST -> {
        if (StringUtils.containsIgnoreCase(e.getMessage(), "duplicate")) {
          yield new PolicyServiceDuplicateException(
              "Policy service throws duplicate object exception", e);
        } else {
          yield new PolicyServiceApiException(e);
        }
      }
      case HttpStatus.SC_CONFLICT ->
          new PolicyConflictException("Policy service throws conflict exception", e);
      default -> new PolicyServiceApiException(e);
    };
```